src/Umbraco.Infrastructure/Packaging/PackageDataInstallation.cs

```diff
@@ -1280,29 +1255,46 @@ public IReadOnlyList<ILanguage> ImportLanguages(IEnumerable<XElement> languageEl
                 _macroService.Save(macro, userId);
             }
 
-            ImportMacroPartialViews(macroPartialViewsElements);
-
             return macros;
         }
 
-        private void ImportMacroPartialViews(IEnumerable<XElement> viewElements)
+        public IReadOnlyList<IPartialView> ImportMacroPartialViews(IEnumerable<XElement> macroPartialViewsElements, int userId)
         {
-            foreach (XElement element in viewElements)
+            var result = new List<IPartialView>();
+
+            foreach (XElement macroPartialViewXml in macroPartialViewsElements)
             {
-                var path = element.AttributeValue<string>("path");
+                var path = macroPartialViewXml.AttributeValue<string>("path");
                 if (path == null)
                 {
                     throw new InvalidOperationException("No path attribute found");
                 }
-                var contents = element.Value ?? string.Empty;
 
-                var physicalPath = _hostingEnvironment.MapPathContentRoot(path);
-                // TODO: Do we overwrite? IMO I don't think so since these will be views a user will change.
-                if (!System.IO.File.Exists(physicalPath))
+                // Remove prefix to maintain backwards compatibility
+                if (path.StartsWith(Constants.SystemDirectories.MacroPartials))
                 {
-                    System.IO.File.WriteAllText(physicalPath, contents, Encoding.UTF8);
+                    path = path.Substring(Constants.SystemDirectories.MacroPartials.Length);
+                }
+                else if (path.StartsWith("~"))
+                {
+                    _logger.LogWarning("Importing macro partial views outside of the Views/MacroPartials directory is not supported: {Path}", path);
+                    continue;
+                }
```

Comment:
This is only required to maintain backwards compatibility with package XMLs that still contain the virtual path.